api/v1alpha1/condition_types.go

```diff
@@ -25,6 +25,14 @@ const (
 	// the progressive sync stages have been completed
 	StagesCompleteReason string = "StagesCompleted"
 
+	// StagesInitializedReason represent the fact that all
+	// the progressive sync stages have been initialized
+	StagesInitializedReason string = "StagesInitialized"
```

Comment:
What is the difference between `StagesInitialized` and `StagesProgressing`? How do we distinguish between them?

Reply:
Indeed, I wasn't sure whether we'd need something like this. StagesInitialized is just the empty/default state with no per stage status, while StagesProgressing was to be set as soon as the controller makes progress on one of the stages. It might be useful in detecting whether the controller is progressing at all, but like I said, I am not sure myself if it is needed right now. Happy to remove it and keep things a bit simpler in the code as well.

Reply:
I would start with just Progressing and then we can expand it in the future.